### PR TITLE
Interchain Queries v2

### PR DESCRIPTION
**Pull request by joe-bowman**
- allow support for all query types - move away from json-encoded params arrays to protobuf-encoded request objects
- support proof based KV store lookups (inclusion and non-inclusion proofs)
- migrate validator and balance queries to KV store lookups
- TODO: delegation lookups
- bump cosmos-sdk (v0.45.4) and tendermint (v0.34.19)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -130,21 +122,23 @@ require (
...
-	gopkg.in/ini.v1 v1.66.3 // indirect
+	gopkg.in/ini.v1 v1.66.4 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
 	nhooyr.io/websocket v1.8.7 // indirect
 )
 
 replace (
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
-	github.com/cosmos/cosmos-sdk => github.com/ingenuity-build/cosmos-sdk v0.45.1-ls-2e
+	//github.com/cosmos/cosmos-sdk => github.com/ingenuity-build/cosmos-sdk v0.45.1-ls-2e
```
Comment (joe-bowman) [inline go.mod:139]: oops, let me fix this.

File: scripts/simple-test.sh [hunk 2]
```diff
@@ -21,8 +21,10 @@ else
   rm -rf ${CHAIN_DIR}/hermes &> /dev/null
   rm -rf ${CHAIN_DIR}/icq &> /dev/null
 
-  TIME=$(date --date '-2 minutes' +%Y-%m-%dT%H:%m:00.00Z -u)
+  TIME=$(date --date '-2 minutes' +%Y-%m-%dT%H:%M:00Z -u)
   jq ".genesis_time = \"$TIME\"" ./${CHAIN_DIR}/backup/${CHAINID_1}/config/genesis.json > ./${CHAIN_DIR}/backup/${CHAINID_1}/config/genesis.json.new && mv ./${CHAIN_DIR}/backup/${CHAINID_1}/config/genesis.json{.new,}
+  jq ".genesis_time = \"$TIME\"" ./${CHAIN_DIR}/backup/${CHAINID_1}a/config/genesis.json > ./${CHAIN_DIR}/backup/${CHAINID_1}a/config/genesis.json.new && mv ./${CHAIN_DIR}/backup/${CHAINID_1}a/config/genesis.json{.new,}
```
Comment (joe-bowman) [inline scripts/simple-test.sh:26]: fixes local test bug where you need to reset state regularly

File: x/interchainquery/keeper/abci.go [hunk 3]
```diff
@@ -31,12 +31,10 @@ func (k Keeper) EndBlocker(ctx sdk.Context) {
 				sdk.NewAttribute(types.AttributeKeyChainId, queryInfo.ChainId),
 				sdk.NewAttribute(types.AttributeKeyConnectionId, queryInfo.ConnectionId),
 				sdk.NewAttribute(types.AttributeKeyType, queryInfo.QueryType),
+				sdk.NewAttribute(types.AttributeKeyHeight, "0"),
```
Comment (joe-bowman) [inline x/interchainquery/keeper/abci.go:34]: TODO: add height to request type.

File: go.mod (hunk 1 above)
Comment (joe-bowman) [reply go.mod]: resolved;

**Review by zanicar — DISMISSED** [review]
lgtm

**Review by zanicar — APPROVED** [review]
lgtm

**Pull request merged**